### ISSUE TITLE
Ref array types as alternative to pointers

### ISSUE THREAD

**Issue by rhenwood39** [cucapra/node-llvmc#1]
In llvm2, ref array types were used in places where this project used pointers (e.g. voidpp, int pointers, etc).

See https://github.com/dirk/llvm2/blob/master/lib/ref-types.js for definition of array types.

See https://github.com/dirk/llvm2/blob/master/lib/library.js for use of array types.

**Comment by rhenwood39** [comment]
It looks like we are going to have to use PointerArrays instead of voidpp and the like. We'll have to go through llvm.ts and see which functions are currently using pointer arguments when they should be using array arguments.

**Comment by sampsyo** [comment]
Yes, good point. I think it's *technically* possible for `voidpp` to work if the user carefully constructs their own Buffer full of pointer-sized numbers, but the new array types are certainly more ergonomic.
